### ISSUE TITLE
Error when installing the core (guava?)

### ISSUE THREAD

**Issue by ElisaRojas** [fp7-netide/Engine#99]
Hi,

A colleague is trying to install the core from scratch and he obtained the following error when installing the core:

```
Error executing command: Can't install feature core/0.0.0:     
Could not start bundle mvn:eu.netide.core/core.management/1.1.0-SNAPSHOT in feature(s) core-management-1.1.0-SNAPSHOT: Uses constraint violation. Unable to resolve bundle revision core.management [150.0] because it is exposed to package 'com.google.common.collect' from bundle revisions com.google.guava [142.0] and com.google.guava [70.0] via two dependency chains.

Chain 1:
  core.management [150.0]
    import: (&(osgi.wiring.package=com.google.common.collect)(version>=18.0.0)(!(version>=19.0.0)))
     |
    export: osgi.wiring.package=com.google.common.collect
  com.google.guava [142.0]

Chain 2:
  core.management [150.0]
    import: (&(osgi.wiring.package=org.onosproject.net)(version>=1.4.0)(!(version>=2.0.0)))
     |
    export: osgi.wiring.package=org.onosproject.net; uses:=org.onlab.packet
  org.onosproject.onos-api [99.0]
    import: (&(osgi.wiring.package=org.onlab.packet)(version>=1.5.0)(!(version>=2.0.0)))
     |
    export: osgi.wiring.package=org.onlab.packet; uses:=com.google.common.collect
  org.onosproject.onlab-misc [97.0]
    import: (&(osgi.wiring.package=com.google.common.collect)(version>=19.0.0)(!(version>=20.0.0)))
     |
    export: osgi.wiring.package=com.google.common.collect
  com.google.guava [70.0]
```

Software installed:

```
apache maven 3.3.9
apache karaf 3.0.5
java 8 oracle
```

Up to this point, we are a bit lost (I had not seen that errors before...). Any clues? @schwabe or @amarsico ?
Thank you!

**Comment by GonzaloFajardoMedina** [comment]
Hello,
we could solve the problem. We had to add Guava dependency in the Pom.xml file within the Core Management folder as follows :

```
< Dependency>
            < GroupId > com.google.guava < / groupId >
            < ArtifactId > guava < / artifactId >
            < Version > r05 < / version>
< / Dependency>
```

Thanks everybody.

**Comment by ElisaRojas** [comment]
Thank you @GonzaloFajardoMedina !  :smile: 

@schwabe , could you check if this solves the issue and could be added to the master branch? (or if there is another way to solve it?)

And then we can close the issue :)

**Comment by schwabe** [comment]
@ElisaRojas I am on vacation today. I will look over it next week.

**Comment by amarsico** [comment]
Hi @ElisaRojas, the solution that you provided is correct. I created a pull request for @schwabe, since I reordered various POMs in the Core. Now they are aligned with the ONOS version 1.5.1. If you are able to test this fix  #101 , please do it. I tested it on Karaf clean and ONOS and it worked correctly.

Best

**Comment by ralvarep** [comment]
Hi @amarsico, I have just tested your pull request #101 and it fixes the problem :+1: 
I merge the pull request with the master branch. Thanks!
